src/lib/utils/options/readers/tsconfig.ts

```diff
@@ -51,31 +51,45 @@ export class TSConfigReader extends OptionsComponent {
             return;
         }
 
-        let data = ts.readConfigFile(fileName, ts.sys.readFile).config;
+        const data = ts.readConfigFile(fileName, ts.sys.readFile).config;
+
         if (data === undefined) {
             event.addError('The tsconfig file %s does not contain valid JSON.', fileName);
             return;
         }
+
         if (!_.isPlainObject(data)) {
             event.addError('The tsconfig file %s does not contain a JSON object.', fileName);
             return;
         }
 
-        data = ts.parseJsonConfigFileContent(
+        let {options, fileNames} = ts.parseJsonConfigFileContent(
             data,
             ts.sys,
             Path.resolve(Path.dirname(fileName)),
             {},
             Path.resolve(fileName));
 
-        event.inputFiles = data.fileNames;
+        event.inputFiles = fileNames;
+        options = _.clone(options);
```

Comment:
Can you also point out where `extends` is now handled? I'm not sure I'm reading this snippet correctly, but it seems to move things around but I can't tell what/where something new was introduced.